torch/fx/passes/infra/partitioner.py

```diff
@@ -165,6 +167,18 @@ def merge_single_node(node: Node, id: Optional[int]):
                     # this is a no-op
                     maybe_merge_partition(self_id, other_id)
 
+        # merge the candidates aggressively
+        if self.aggressive_merge:
+            merge_candidates_list = list(set(assignment.values()))
+            while True:
+                size = len(merge_candidates_list)
+                self_id = merge_candidates_list[0]
+                for other_id in merge_candidates_list[1:]:
+                    maybe_merge_partition(self_id, other_id)
```

Comment:
I think this is a bug in the original impl. This case should have been handled in line 168.

Reply:
Sure. I can do that.

Reply:
@SherlockNoMad I fixed the merging logic. Please take a look and enable the CI. Thanks.

Reply:
Thank you for the fix!